sympy/integrals/rationaltools.py

```diff
@@ -52,8 +52,12 @@ def ratint(f, x, **flags):
     else:
         p, q = f.as_numer_denom()
 
+    p = p.xreplace({c: Rational(c) for c in p.atoms(Float)})
+    q = q.xreplace({c: Rational(c) for c in q.atoms(Float)})
```

Comment:
This is not the right way to do this. Rather somewhere there should be a domain conversion.